libs/sdk-ui-ext/src/insightView/InsightRenderer.tsx

```diff
@@ -112,31 +112,40 @@ class InsightRendererCore extends React.PureComponent<IInsightRendererProps & Wr
         // the visualization we may have from earlier is no longer valid -> get rid of it
         this.unmountVisualization();
 
-        const visualizationFactory = FullVisualizationCatalog.forInsight(this.props.insight);
-
-        this.visualization = visualizationFactory({
-            backend: this.props.backend,
-            callbacks: {
-                onError: (error) => {
-                    this.props.onError?.(error);
-                    this.props.onLoadingChanged?.({ isLoading: false });
-                },
-                onLoadingChanged: ({ isLoading }) => {
-                    this.props.onLoadingChanged?.({ isLoading });
+        let visualizationFactory;
+
+        try {
+            visualizationFactory = FullVisualizationCatalog.forInsight(this.props.insight).getFactory();
+        } catch (e) {
+            // eslint-disable-next-line no-console
+            console.error(`Error: unsupported visualization type - ${this.props.insight}`);
```

Comment:
The error message seems to reference too broad an object. Should we use the visualization type?